lib/events.js

```diff
@@ -214,50 +226,56 @@ EventEmitter.prototype.addListener = function addListener(type, listener) {
     existing = events[type];
   }
 
+  // Handle custom this object if given
+  typeof self === 'undefined' ||
+    (listener = { listener: listener, self: self });
```

Comment:
It would be clearer with `if (typeof self !== 'undefined') { ... }` IMO.
This kind of trick is more suited for code minifiers.